cmd/zar/zq/command.go

```diff
@@ -0,0 +1,238 @@
+	outputFile     string
+	stopErr        bool
+	quiet          bool
+	ReaderFlags    zio.ReaderFlags
+	WriterFlags    zio.WriterFlags
+}
+
+func fileExists(path string) bool {
+	if path == "-" {
+		return true
+	}
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+	return !info.IsDir()
+}
+
+func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &Command{Command: parent.(*root.Command)}
+	f.StringVar(&c.root, "R", os.Getenv("ZAR_ROOT"), "root directory of zar archive to walk")
+	f.BoolVar(&c.quiet, "q", false, "don't display zql warnings")
+	// XXX f.StringVar(&c.dir, "d", "", "directory for output data files")
+	f.StringVar(&c.outputFile, "o", "", "write data to output file")
+	f.StringVar(&c.jsonTypePath, "j", "", "path to json types file")
+	f.StringVar(&c.jsonPathRegexp, "pathregexp", c.jsonPathRegexp, "regexp for extracting _path from json log name (when -inferpath=true)")
+	// XXX f.BoolVar(&c.quiet, "q", false, "don't display zql warnings")
+	f.BoolVar(&c.stopErr, "e", true, "stop upon input errors")
+
+	// Flags added for writers are -f, -T, -F, -E, -U, and -b
+	c.WriterFlags.SetFlags(f)
+
+	// Flags added for readers are -i XXX json
+	c.ReaderFlags.SetFlags(f)
+
+	return c, nil
+}
+
+func (c *Command) loadJsonTypes() (*ndjsonio.TypeConfig, error) {
+	data, err := ioutil.ReadFile(c.jsonTypePath)
+	if err != nil {
+		return nil, err
+	}
+	var tc ndjsonio.TypeConfig
+	err = json.Unmarshal(data, &tc)
+	if err != nil {
+		return nil, fmt.Errorf("%s: unmarshaling error: %s", c.jsonTypePath, err)
+	}
+	if err = tc.Validate(); err != nil {
+		return nil, fmt.Errorf("%s: %s", c.jsonTypePath, err)
+	}
+	return &tc, nil
+}
+
+//XXX lots here copied from zq command... we should refactor into a tools package
+func (c *Command) Run(args []string) error {
+	//XXX
+	if c.outputFile == "-" {
+		c.outputFile = ""
+	}
+
+	if _, err := regexp.Compile(c.jsonPathRegexp); err != nil {
+		return err
+	}
+	if c.jsonTypePath != "" {
+		tc, err := c.loadJsonTypes()
+		if err != nil {
+			return err
+		}
+		c.jsonTypeConfig = tc
+	}
+	if len(args) == 0 {
+		return errors.New("zar zq needs input arguments")
+	}
+	rootPath := c.root
+	if rootPath == "" {
+		rootPath = "."
+	}
+	// XXX this is parallelizable except for writing to stdout when
+	// concatenating results
+	return archive.Walk(rootPath, func(zardir string) error {
+		inputs := args
+		var query ast.Proc
+		var err error
+		first := archive.Localize(zardir, inputs[:1])
+		if first[0] != "" && fileExists(first[0]) {
+			query, err = zql.ParseProc("*")
+			if err != nil {
+				return err
+			}
+		} else {
+			query, err = zql.ParseProc(inputs[0])
+			if err != nil {
+				return err
+			}
+			inputs = inputs[1:]
+		}
+		localPaths := archive.Localize(zardir, inputs)
+		readers, err := c.inputReaders(resolver.NewContext(), localPaths)
+		if err != nil {
+			return err
+		}
+		if len(readers) == 0 {
+			// XXX silently skip if no file found
```

Comment:
Skipping & proceeding seems like the right thing to do; do you think we should log here?

Reply:
changing to warn to stderr in no -q